sdk/openai/openai/samples/cookbook/bringYourOwnData/index.js

```diff
@@ -0,0 +1,124 @@
+        return;
+    }
+
+    // Creating an index client to create the search index
+    const indexClient = new SearchIndexClient(azureSearchEndpoint, new AzureKeyCredential(azureSearchAdminKey));
+
+    // Getting the name of the index from the index definition
+    const indexName = indexDefinition["name"];
+
+    console.log('Checking if index exists...');
+    await deleteIndexIfExists(indexClient, indexName);
+
+    console.log('Creating index...');
+    let index = await indexClient.createIndex(indexDefinition);
+    console.log(`Index named ${index.name} has been created.`);
+
+    // Creating a search client to upload documents and issue queries
+    const searchClient = indexClient.getSearchClient(indexName);
+
+    console.log('Uploading documents...');
+    let indexDocumentsResult = await searchClient.mergeOrUploadDocuments(hotelData['value']);
+    console.log(`Index operations succeeded: ${JSON.stringify(indexDocumentsResult.results[0].succeeded)} `);
+
+    // waiting one second for indexing to complete (for demo purposes only)
+    sleep(1000);
+
+    // send example question to open AI instance
+    const exampleQuestions = [
+      {
+        role: "user",
+        content: "Which hotel would you recommend for a foodie, and why?",
+      },
+    ];
+
+    console.log(`Asking ChatGPT: ${exampleQuestions[0].content}`);
+    console.log();
+    await askOpenAI(indexName, exampleQuestions);
+}
+
+async function deleteIndexIfExists(indexClient, indexName) {
+    try {
+        await indexClient.deleteIndex(indexName);
+        console.log('Deleting index...');
+    } catch {
+        console.log('Index does not exist yet.');
+    }
+}
+
+async function askOpenAI(azureSearchIndexName, messages) {
+    const client = new OpenAIClient(openAiEndpoint, new AzureKeyCredential(azureApiKey));
+    const events = client.listChatCompletions(chatDeploymentId, messages, {
+      maxTokens: 128,
+      /**
+       * The `azureExtensionOptions` property is used to configure the
+       * Azure-specific extensions. In this case, we are using the
+       * Azure Cognitive Search extension with a vector index to provide
+       * the model with additional context.
+       */
+      azureExtensionOptions: {
+        extensions: [
+          {
+            type: "AzureCognitiveSearch",
+            parameters: {
+              endpoint: azureSearchEndpoint,
+              key: azureSearchAdminKey,
+              indexName: azureSearchIndexName,
+            },
+          },
+        ],
+      },
+    });
+
+    let chatGptAnswer = "";
+    for await (const event of events) {
+      for (const choice of event.choices) {
+        const newText = choice.delta?.content;
+        if (!!newText) {
+          chatGptAnswer += newText;
+          // To see streaming results as they arrive, uncomment line below
+          // console.log(newText);
+        }
+      }
+    }
+    console.log(chatGptAnswer);
+}
+
+function sleep(ms) {
```

Comment:
Are we forcing a busy wait in Node.js with blocking?  Why can't we use Promises or timeouts?

Reply:
Updated with [`90f9611`](https://github.com/Azure/azure-sdk-for-js/pull/27298/commits/90f9611a8f53cd98e2ca46f12c1bfbd931261293)